File: Java/Sorpresa_PorJugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ilzk
 */
public class Sorpresa_PorJugador extends Sorpresa {
    
    public Sorpresa_PorJugador(){
        valor = 10;
    }
    
    @Override
    /**Aplica la sorpresa Por jugador al jugador actual
     * 
     */
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual,todos);
            Sorpresa quitar = new Sorpresa_PagarCobrar();
            quitar.valor = -valor;
                for (int i = 0; i < todos.size(); i++) {
                    if (i != actual)
                        quitar.aplicarAJugador(i, todos);
                }
            Sorpresa dar = new Sorpresa_PagarCobrar();
            dar.valor = valor * todos.size();
            dar.aplicarAJugador(actual, todos);
        }
    }
    
    /**Sobreescribe el metodo toString
     * 
     */
    public String toString(){
        return "Por jugador";
    }
}

File: Ruby/sorpresa_por_jugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'sorpresa'
module Civitas
  class Sorpresa_por_jugador < Sorpresa
    attr
    def initialize
      @valor = 10
    end

    def aplicar_a_jugador(actual, todos)
      if (jugador_correcto(actual, todos))
          informe(actual,todos)
          quitar = Sorpresa_pagar_cobrar.new()
          quitar.valor = 0-@valor
          for i in 1..todos.size
            if (i != actual)
              quitar.aplicar_a_jugador(i, todos)
            end
          end
          dar = Sorpresa_pagar_cobrar.new()
          dar.valor = todos.size * @valor
          dar.aplicar_a_jugador(actual, todos)
      end
    end

    def to_s()
      "Por jugador"
    end
    public_class_method:new
  end
end